test/controllers/journeys/income/IncomeCYAControllerSpec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.journeys.income
+
+import base.SpecBase
+import controllers.journeys.income.routes.IncomeCYAController
+import models.{Income, NormalMode, UserAnswers}
+import play.api.i18n.Messages
+import play.api.libs.json.{JsObject, Json}
+import play.api.mvc.Result
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryList
+import viewmodels.checkAnswers.income._
+import views.html.journeys.income.IncomeCYAView
+
+import scala.concurrent.Future
+
+class IncomeCYAControllerSpec extends SpecBase {
+
+  private val authUserType = "individual"
+  private val businessId = "some_business_id"
+
+  private val userAnswerData = Json.parse(s"""
+       |{
+       |  "incomeNotCountedAsTurnover": false,
+       |  "turnoverIncomeAmount": 1234.55,
+       |  "anyOtherIncome": false,
+       |  "turnoverNotTaxable": false,
+       |  "tradingAllowance": "declareExpenses"
+       |}
+       |""".stripMargin)
+
+  private val userAnswers = UserAnswers("some_id", userAnswerData.as[JsObject])
+
```

Comment:
Should probably add user scenarios for Welsh/English, User/Individual, and for different combos of user answers to make sure the view correctly edits for SummaryList for a different number of rows